omc3/scripts/check_peak_to_peak.py

```diff
@@ -0,0 +1,199 @@
+
+import numpy as np
+import pandas as pd
+import turn_by_turn as tbt
+from generic_parser import EntryPointParameters, entrypoint
+from turn_by_turn.utils import generate_average_tbtdata
+
+from omc3.definitions.constants import PLANES, UNIT_IN_METERS
+from omc3.harpy.constants import DEFOCUSSING_MONITORS
+from omc3.hole_in_one import HARPY_DEFAULTS
+from omc3.tbt_converter import _file_name_without_sdds
+from omc3.utils.logging_tools import get_logger
+
+LOG = get_logger(__name__)
+
+
+def get_params():
+    params = EntryPointParameters()
+    params.add_parameter(name="files", required=True, nargs='+',
+                         help="Files for analysis")
+    params.add_parameter(name="beam", type=int,
+                         help="LHC beam number.")
+    params.add_parameter(name="tbt_datatype",
+                         default=HARPY_DEFAULTS["tbt_datatype"],
+                         choices=list(tbt.io.DATA_READERS.keys()),
+                         help="Choose the datatype from which to import.")
+    params.add_parameter(name="unit", type=str, default="mm",
+                         choices=list(UNIT_IN_METERS.keys()),
+                         help="Unit in which to log the peak-to-peak values.")
+    params.add_parameter(name="input_unit", type=str, default="mm",
+                         choices=list(UNIT_IN_METERS.keys()),
+                         help="Unit of the tbt input data.")
+    return params
+
+
+@entrypoint(get_params(), strict=True)
+def peak_to_peak(opt):
+    """Main function to log peak-to-peak values from SDDS files.
+
+    *--Required--*
+
+    - **files**:
+
+        Files for analysis
+
+
+    *--Optional--*
+
+    - **beam** *(int)*:
+
+        LHC beam number.
+
+
+    - **input_unit** *(str)*:
+
+        Unit of the tbt input data.
+
+        choices: ``['km', 'm', 'mm', 'um', 'nm', 'pm', 'fm', 'am']``
+
+        default: ``mm``
+
+
+    - **tbt_datatype**:
+
+        Choose the datatype from which to import.
+
+        choices: ``['lhc', 'iota', 'esrf', 'ptc', 'trackone']``
+
+        default: ``lhc``
+
+
+    - **unit** *(str)*:
+
+        Unit in which to log the peak-to-peak values.
+
+        choices: ``['km', 'm', 'mm', 'um', 'nm', 'pm', 'fm', 'am']``
+
+        default: ``mm``
+
+    """
+    for input_file in opt.files:
+        LOG.debug(f"Calculating pk2pk for file: {input_file}")
+        input_file = Path(input_file)
+        name = _file_name_without_sdds(input_file)
+        beam = _get_beam(opt.beam, filename=name)
+        bpms = DEFOCUSSING_MONITORS[beam]
+        tbt_data = tbt.read_tbt(input_file, datatype=opt.tbt_datatype)
+        for bunch, positions in zip(tbt_data.bunch_ids, tbt_data.matrices):
+            LOG.debug(f"Bunch: {bunch}")
+            pk2pk = get_pk2pk(positions, bpms)
+            _log_pk2pk(pk2pk, name, bunch, opt.input_unit, opt.unit)
+
+        if tbt_data.nbunches > 1:
+            tbt_data_av = generate_average_tbtdata(tbt_data)
```

Comment:
Does Multiturn generate an average of the bunches too? If not, this could be why you observe a difference.

Reply:
I only generate an average if there is more than one bunch in the file. and then also additionally to the per bunch pk2pk. And for the tested files, there is only a single bunch. So no idea what the difference is